src/Microsoft.TestPlatform.Build/Tasks/TestTaskUtils.cs

```diff
@@ -0,0 +1,222 @@
+        }
+
+        builder.AppendSwitchIfNotNull("--diag:", task.VSTestDiag);
+
+        if (task.TestFileFullPath == null || task.TestFileFullPath.ItemSpec.IsNullOrEmpty())
+        {
+            task.Log.LogError(Resources.Resources.TestFilePathCannotBeEmptyOrNull);
+        }
+        else
+        {
+            builder.AppendFileNameIfNotNull(task.TestFileFullPath);
+        }
+
+        // Console logger was not specified by user, but verbosity was, hence add default console logger with verbosity as specified
+        if (!task.VSTestVerbosity.IsNullOrWhiteSpace() && !isConsoleLoggerSpecifiedByUser)
+        {
+            var normalTestLogging = new List<string>() { "n", "normal", "d", "detailed", "diag", "diagnostic" };
+            var quietTestLogging = new List<string>() { "q", "quiet" };
+
+            string vsTestVerbosity = "minimal";
+            if (normalTestLogging.Contains(task.VSTestVerbosity.ToLowerInvariant()))
+            {
+                vsTestVerbosity = "normal";
+            }
+            else if (quietTestLogging.Contains(task.VSTestVerbosity.ToLowerInvariant()))
+            {
+                vsTestVerbosity = "quiet";
+            }
+
+            builder.AppendSwitchUnquotedIfNotNull("--logger:", $"Console;Verbosity={vsTestVerbosity}");
+        }
+
+        if (task.VSTestBlame || task.VSTestBlameCrash || task.VSTestBlameHang)
+        {
+            var dumpArgs = new List<string>();
+            if (task.VSTestBlameCrash || task.VSTestBlameHang)
+            {
+                if (task.VSTestBlameCrash)
+                {
+                    dumpArgs.Add("CollectDump");
+                    if (task.VSTestBlameCrashCollectAlways)
+                    {
+                        dumpArgs.Add($"CollectAlways={task.VSTestBlameCrashCollectAlways}");
+                    }
+
+                    if (!task.VSTestBlameCrashDumpType.IsNullOrEmpty())
+                    {
+                        dumpArgs.Add($"DumpType={task.VSTestBlameCrashDumpType}");
+                    }
+                }
+
+                if (task.VSTestBlameHang)
+                {
+                    dumpArgs.Add("CollectHangDump");
+
+                    if (!task.VSTestBlameHangDumpType.IsNullOrEmpty())
+                    {
+                        dumpArgs.Add($"HangDumpType={task.VSTestBlameHangDumpType}");
+                    }
+
+                    if (!task.VSTestBlameHangTimeout.IsNullOrEmpty())
+                    {
+                        dumpArgs.Add($"TestTimeout={task.VSTestBlameHangTimeout}");
+                    }
+                }
+            }
+
+            if (dumpArgs.Any())
+            {
+                builder.AppendSwitchIfNotNull("--Blame:", string.Join(";", dumpArgs));
+            }
+            else
+            {
+                builder.AppendSwitch("--Blame");
+            }
+        }
+
+        if (task.VSTestCollect != null && task.VSTestCollect.Any())
```

Comment:
The check to `Any` is not needed.

```suggestion
        if (task.VSTestCollect != null)
```